client/hosting-overview/components/plan-card.tsx

```diff
@@ -0,0 +1,119 @@
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+const PlanCard: FC = () => {
+	const moment = useLocalizedMoment();
+	const site = useSelector( getSelectedSite );
+	const sitePlan = site?.plan;
+	const planName = sitePlan?.product_name_short ?? '';
+	const planSlug = ( sitePlan?.product_slug || '' ) as PlanSlug;
+	const plan = useSelector( ( state ) => getCurrentPlan( state, site?.ID ) );
+	const pricing = usePricingMetaForGridPlans( {
+		coupon: undefined,
+		planSlugs: [ planSlug ],
+		selectedSiteId: site?.ID,
+		storageAddOns: null,
+		useCheckPlanAvailabilityForPurchase,
+	} );
+	const translate = useTranslate();
+
+	// TODO: Replace with real data using <PlanStorage /> component
+	const usedGigabytes = 47;
+	const availableUnitAmount = 50;
+
+	return (
+		<>
+			<QuerySitePlans siteId={ site?.ID } />
+			<Card className="hosting-overview__card">
+				<div>
+					<div className="hosting-overview__plan-card-header">
+						<h3 className="hosting-overview__plan-card-title">{ planName }</h3>
+
+						<Button
+							className="hosting-overview__link-button"
+							plain
+							href={ `/plans/${ site?.slug }` }
+						>
+							{ translate( 'Manage plan' ) }
+						</Button>
+					</div>
+					<PlanPrice
+						className="hosting-overview__plan-price"
+						currencyCode={ plan?.currencyCode }
+						displayPerMonthNotation
+						isSmallestUnit
+						rawPrice={ pricing?.[ planSlug ].originalPrice.monthly }
+					/>
+					<div className="hosting-overview__plan-info">
+						{ translate( '{{span}}%(rawPrice)s{{/span}} billed annually.', {
+							args: {
+								rawPrice: formatCurrency(
+									pricing?.[ planSlug ].originalPrice.full ?? 0,
+									plan?.currencyCode ?? '',
+									{
+										stripZeros: true,
+										isSmallestUnit: true,
+									}
+								),
+							},
+							components: {
+								span: <span />,
+							},
+						} ) }
+					</div>
+					<div className="hosting-overview__plan-info">
+						{ translate( 'Expires on %s.', { args: moment( plan?.expiryDate ).format( 'LL' ) } ) }
+					</div>
+					<div className="hosting-overview__plan-storage">
+						<div className="hosting-overview__plan-storage-title-wrapper">
+							<div className="hosting-overview__plan-storage-title">
+								<Icon icon={ cloud } />
+								{ translate( 'STORAGE' ) }
+							</div>
+							<span>
+								{ translate(
+									'Using {{usedStorage}}%(usedGigabytes).1fGB{{/usedStorage}} of %(availableUnitAmount)dGB',
+									'Using {{usedStorage}}%(usedGigabytes).1fGB{{/usedStorage}} of %(availableUnitAmount)dGB',
+									{
+										count: usedGigabytes,
+										args: { usedGigabytes, availableUnitAmount },
+										comment:
+											'Must use unit abbreviation; describes used vs available storage amounts (e.g., Using 20.0GB of 30GB, Using 0.5GB of 20GB)',
+										components: { usedStorage: <span className="used-space__span" /> },
+									}
+								) }
+							</span>
```

Comment:
I'm seeing wrong data on my free site (Free plans have 1GB of storage):

<img width="417" alt="Screenshot 2024-04-12 at 14 24 47" src="https://github.com/Automattic/wp-calypso/assets/1233880/7225c361-d20a-4f78-9563-6dd49508a9b5">

Reply:
Yes, I forgot to mention it on the description. I'll replace it with real data using the `<PlanStorage />` component on a followup PR, because that component will need some tweaking.